src/nom.rs

```diff
@@ -602,9 +602,10 @@ pub fn float(input: &[u8]) -> IResult<&[u8],f32> {
     recognize!(
       tuple!(
         opt!(alt!(tag!("+") | tag!("-"))),
-        alt!(
-          delimited!(digit, tag!("."), opt!(digit))
-          | delimited!(opt!(digit), tag!("."), digit)
+        alt_complete!(
+          delimited!(opt!(digit), tag!("."), digit)
+          | preceded!(digit, tag!("."))
+          | digit
```

Comment:
That's basically the same as https://github.com/Geal/nom/blob/master/benches/json.rs#L27 now, just as a general comment. Change looks good to me!